### PR TITLE
Improved UI/UX of the Attributes tab (v2)

### LINKED ISSUE #37839: [Epic] Attributes tab enhancements v2
I've reviewed the changes implemented in 7.7 and came up with a few minor improvements we can make to streamline the experience further.

**Design**

`zs56kcy53K7GoGhgyjcQ2R-fi-365-39439&t=IlDQFEuvZsBvigFP-4`

image

**Acceptance criteria**

- [x] #38104
- [ ] We automatically select all values when the user adds an existing attribute.
- [ ] When adding an existing attribute, the placeholder text in the Value/s field reads `Select values` (currently, it's `Select terms`).
- [ ] When the `Used for variations` box is unchecked, the placeholder text in the `Value/s` field reads `Enter some descriptive text. Use “|” to separate different values.`
- [ ] The notice in the empty state is dismissible. Once hidden, it does not re-appear.

### PR DESCRIPTION
**Pull request by jimjasson**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR implements the design improvements for the Attributes that were suggested by @jarekmorawski: 

image

Specifically: 
- It makes the notice that shows up in the Attributes tab is dismissible and once dismissed, it ensures that it won't show up again. 
- It automatically selects all the terms when a global Attribute is chosen. 
- It updates the placeholder text for global Attributes. 
- It conditionally displays a different placeholder text for product-level Attributes, depending on the value of the "Used for Variations" checkbox.
- It fixes the styling of the individual attribute elements -- before the PR, they had a double-border. 

Closes #37839 

### Simple Products

1. Create a global Attribute with some terms, under the **Products > Attributes** tab.
2. Create a Simple Product.
3. Navigate to the **Product Data > Attributes** tab.
4. Ensure that an onboarding notice shows up at the top of the tab, and the notice is dismissible.
5. Click the 'x' button to dismiss the notice. 
6. Refresh the page. 
7. Navigate to the **Product Data > Attributes** tab.
9. Ensure that the onboarding notice is now hidden.
10. Select the global Attribute you previously created: 
11. Ensure that all the Attribute terms are automatically selected. 
12. Remove all the terms and ensure that the placeholder of the "Value(s)" field reads: "Select values".
13. Ensure that the border around the Attribute element is correct (and there isn't a double-border) -- see: https://github.com/woocommerce/woocommerce/issues/37839#issuecomment-1597221883
14. Create a new, product-level Attribute: 
15. Ensure that the placeholder of the "Value(s)" field reads: "Enter some descriptive text. Use “|” to separate different values."

### Variable Products

1. Repeat steps 7-13. 
2. Ensure that when the "Used for variations" checkbox is checked in a product-level Attribute, the placeholder of the "Value(s)" field reads: "Enter options for customers to choose from, f.e. “Blue” or “Large”. Use “|” to separate different options." 
3. When this checkbox is not checked, the placeholder of the "Value(s)" field reads: "Enter some descriptive text. Use “|” to separate different values."

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [x] Enhancement - Improvement to existing functionality

#### Message 

Design enhancements for the Attributes tab.

#### Comment 

</details>

### CODE REVIEW

**Comment by jimjasson** [comment]
Updated the e2e tests, which grabbed the "Values" `textarea` element by its placeholder. However, with this PR the placeholder for this element is added by JS, only when the `textarea` is empty. If it is filled in, then the placeholder is not added. So, we could grab this element by label, which is more robust.

File: plugins/woocommerce/client/legacy/css/admin.scss
Comment (octaedro) [inline plugins/woocommerce/client/legacy/css/admin.scss]: I think that this code is adding an unexpected behavior.
It's removing the line between attributes:

Screenshot 2023-09-01 at 12 24 51

File: plugins/woocommerce/includes/admin/class-wc-admin-assets.php
Comment (octaedro) [inline plugins/woocommerce/includes/admin/class-wc-admin-assets.php]: Can we add these lines directly to [this array](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/admin/class-wc-admin-assets.php#L371-L440)?

**Review by octaedro — COMMENTED** [review]
Nice job @jimjasson! The code looks mainly good and is testing well here. I just left a couple of small comments.

File: plugins/woocommerce/client/legacy/css/admin.scss
Comment (jimjasson) [reply plugins/woocommerce/client/legacy/css/admin.scss]: Spot on!

File: plugins/woocommerce/includes/admin/class-wc-admin-assets.php
Comment (jimjasson) [reply plugins/woocommerce/includes/admin/class-wc-admin-assets.php]: I created a parameters array specifically for the `meta-boxes-product.js` script, however, I see that this script also uses some of the parameters from the `wc-admin-meta-boxes` script, so I can definitely move these two strings to the other array too.

**Comment by jimjasson** [comment]
Thank you for the review @octaedro ! I addressed the two points you mentioned -- could you have another look, please? Thanks!

**Review by octaedro — APPROVED** [review]
Thank you @jimjasson for addressing the changes. LGTM 🚀
Could you merge this PR and https://github.com/woocommerce/woocommerce/pull/39962?

**Comment by jimjasson** [comment]
Awesome, thanks @octaedro -- I was waiting for a review from the designer (@jarekmorawski) but looks like he is AFK for a while, so perhaps it is best to merge this and re-iterate if he has any comments.

**Pull request merged**
